scripts/partition_manager.py

```diff
@@ -24,12 +24,42 @@ def item_is_placed(d, item, after_or_before):
            d['placement'][after_or_before][0] == item
 
 
+def fix_one_of(reqs, partitions):
+    for k, v in reqs.items():
+        if isinstance(v, dict):
+            if 'one_of' in v.keys():
+                if len(v.keys()) != 1:
+                    raise RuntimeError("'one_of' must be the only key in its dict")
+                # Now fetch the first existing partition. Note that the value must be a list even if there is only
+                # one entry.
+                reqs[k] = [partition for partition in partitions if partition in v['one_of']][:1]
+            else:
+                fix_one_of(v, partitions)
+        # 'one_of' dicts can occur inside lists of partitions.
+        # dicts with 'one_of' key is the only dict supported inside lists
+        elif isinstance(v, list):
+            # We need thees as one should not modify the list being iterated
+            to_remove = list()
+            to_add = list()
+            for i in v:
+                if isinstance(i, dict):
+                    if 'one_of' not in i.keys():
+                        raise RuntimeError("Found illegal dict inside list. Only 'one_of' dicts are allowed")
+                    to_add.append([partition for partition in partitions if partition in i['one_of']][0])
```

Comment:
Why not

    to_add.append(i['one_of'][0])

?

Reply:
It fails the test, will look into it.

Reply:
This is not correct, as it does not check for the existence of the item in the list.

Reply:
Ah of course :D

I actually think it's incorrect though. If `partitions` is `['foo', 'bar']` and `i['one_of']` is `['bar', 'foo']`. the result will be 'foo', but should be 'bar'?

Reply:
So it should be the inverse, so that we iterate over the items in `i` and check if they are in `partitions`

Reply:
Good catch, I fixed this for both list and dicts, and added test cases.